src/controllers/mainController.ts

```diff
@@ -378,33 +380,47 @@ export default class MainController implements vscode.Disposable {
 
     /**
      * Called by VS Code when a text document closes. This will dispatch calls to other
-     * controllers as needed. Determines if this was a closed file or if it was an instance
-     * where a file was saved to disk after being an untitled file.
+     * controllers as needed. Determines if this was a normal closed file, a untitled closed file,
+     * or a renamed file
      * @param doc The document that was closed
      */
     private onDidCloseTextDocument(doc: vscode.TextDocument): void {
         let closedDocumentUri: string = doc.uri.toString();
         let closedDocumentUriScheme: string = doc.uri.scheme;
 
-        // Did we save a document before this close event? Was it an untitled document?
-        if (this._lastSavedUri && this._lastSavedTimer && closedDocumentUriScheme === Constants.untitledScheme) {
-            // Stop the save timer
+        // Stop timers if they have been started
+        if (this._lastSavedTimer) {
             this._lastSavedTimer.end();
+        }
 
-            // Check that we saved a document *just* before this close event
-            // If so, then we saved an untitled document and need to update where necessary
-            if (this._lastSavedTimer.getDuration() < Constants.untitledSaveTimeThreshold) {
-                this._connectionMgr.onUntitledFileSaved(closedDocumentUri, this._lastSavedUri);
-            }
+        if (this._lastOpenedTimer) {
+            this._lastOpenedTimer.end();
+        }
+
+        // Determine which event caused this close event
+        if (this._lastSavedUri &&
+                closedDocumentUriScheme === Constants.untitledScheme &&
+                this._lastSavedTimer.getDuration() < Constants.untitledSaveTimeThreshold) {
+            // Untitled file was saved
+            this._connectionMgr.onUntitledFileSaved(closedDocumentUri, this._lastSavedUri);
+
+        } else if (this._lastOpenedUri &&
+                this._lastOpenedTimer.getDuration() < Constants.renamedOpenTimeThreshold) {
```

Comment:
A file rename event consists of an openTextDocument event immediately followed by a closeTextDocument event. So this condition is basically saying if the last event which occurred was an openTextDocument event and it happened just before the this closeTextDocument event then the file must have been renamed.